net-dialup/radiusclient/files/radiusclient-0.3.2-BJA-cross.diff

```diff
@@ -0,0 +1,82 @@
+From: Bertrand Jacquin <bertrand@jacquin.bzh>
+Date: Tue Jul 26 20:17:14 2011 +0200
+Subject: Fix cross-compile
+
+There is no fallback method to determine field domainname in struct utsname:
+
+  checking for field domainname in struct utsname... configure: error: in `/usr/i586-geode-linux-gnu/tmp/portage/net-dialup/radiusclient-0.3.2/work/radiusclient-0.3.2': 
+  configure: error: cannot run test program while cross compiling 
+  See `config.log' for more details
+
+Let the user define utsname status by defining ac_cv_struct_utsname
+
+/dev/urandom is also tested on the build host, which can be different that the
+target host, give option to define ac_cv_file__dev_urandom
+
+--- configure.in	2002-02-03 14:51:40.000000000 +0100
++++ configure.in	2011-06-28 14:27:20.000000000 +0200
+@@ -89,32 +89,40 @@
+ AC_CHECK_FUNCS(flock fcntl uname gethostname sysinfo getdomainname)
+ AC_CHECK_FUNCS(stricmp random rand snprintf vsnprintf)
+ 
+-if test "$ac_cv_func_uname" = 'yes'
+-then
+-	AC_MSG_CHECKING([for field domainname in struct utsname])
+-	AC_TRY_RUN([
+-	#include <sys/utsname.h>
++AC_CACHE_CHECK([for field domainname in struct utsname], ac_cv_struct_utsname,
++ [AC_TRY_RUN([
```

Comment:
Are you sure you are using `AC_TRY_COMPILE` correctly? Note that the arguments are different than for `AC_TRY_RUN`.

```
 -- Macro: AC_TRY_COMPILE (INCLUDES, FUNCTION-BODY, [ACTION-IF-FOUND [,
          ACTION-IF-NOT-FOUND]])

 -- Macro: AC_TRY_RUN (PROGRAM, [ACTION-IF-TRUE [, ACTION-IF-FALSE [,
          ACTION-IF-CROSS-COMPILING]]])